### PR TITLE
New utility for splitting mzML files

### PR DESCRIPTION
**Pull request by hendrikweisser**
Utility for splitting mzML files into multiple parts, optionally leaving out chromatograms or spectra. Useful for splitting files that are too big to search easily (for example after merging results from multiple fractions).

### CODE REVIEW

**Comment by hroest** [comment]
This seems like a nice tool but can you explain the workflow where this may be used? I am wondering about your usecase where you say that you first merge results from multiple fractions and then split them again using this tool -- wouldnt it be easier then to keep the fractions separate rather than merging and splitting them in this manner?

Also, if you are worried about files getting too large to handle, have you thought about using some of the low memory tools and indexed mzML functionality that may make your life easier? It may also be worth a shot here as these files are potentially large (given that you want to split them) and you may benefit from these functions.

File: src/utils/MzMLSplitter.cpp [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    setValidFormats_("in", ListUtils::create<String>("mzML"));
+    registerStringOption_("out", "<file>", "", "Prefix for output files ('_part1.mzML' etc. will be appended; default: same as 'in' without the file extension)", false);
+    registerIntOption_("parts", "<num>", 1, "Number of parts to split into (takes precedence over 'size' if set)", false);
+    setMinInt_("parts", 1);
+    registerIntOption_("size", "<num>", 0, "Approximate upper limit for resulting file sizes (in 'unit')", false);
+    setMinInt_("size", 0);
+    registerStringOption_("unit", "<choice>", "MB", "Unit for 'size' (base 1024)", false);
+    setValidStrings_("unit", ListUtils::create<String>("KB,MB,GB"));
+    // @TODO:
+    // registerFlag_("precursor", "Make sure precursor spectra end up in the same part as their fragment spectra");
```
Comment (hroest) [inline src/utils/MzMLSplitter.cpp]: seems like this should be done by default, is this is almost always the only sensible thing to do

**Comment by hroest** [comment]
seems like some of the new tests dont behave well

http://cdash.openms.de/viewTest.php?onlyfailed&buildid=147263

**Comment by hendrikweisser** [comment]
> This seems like a nice tool but can you explain the workflow where this may be used?

We are doing proteogenomics on tissue data. When we run searches, we can't have too few spectra in a file - we need a few thousand IDs for Percolator to work reliably. However, we also can't have too many spectra per file - there's a hard limit of 2 GB for the MGF file produced in MascotAdapterOnline, or we can't transmit it over HTTP to the Mascot server using the current Qt mechanism. Moreover, "large" files (where it's unclear how large "large" is) are currently causing us trouble in Mascot 2.5 (see #1624).

So when we have a set of raw data files from a fractionated sample, with varying sizes/numbers of (MS2) spectra, we can either merge groups of them until we get roughly the sizes we want, or we can merge all and then split to the right size. I think the second is a bit easier, especially if we have to experiment with what "the right size" is.

> Also, if you are worried about files getting too large to handle, have you thought about using some of the low memory tools and indexed mzML functionality that may make your life easier?

No, since the files that can get too large are the MGFs produced in MascotAdapterOnline.

In any case, while this tool may not be necessary for many people, I still think it's a useful addition to the functionality offered by OpenMS.

**Comment by hendrikweisser** [comment]
> seems like some of the new tests dont behave well

I forgot to define some dependencies between the tests that generate output files and those that compare them. I hope adding those dependencies fixes the problems. We'll see.

**Comment by timosachsenberg** [comment]
We have something similar for ID files. There there tool is called IDRipper.

File: src/utils/MzMLSplitter.cpp (hunk 1 above)
Comment (hendrikweisser) [inline src/utils/MzMLSplitter.cpp]: I thought about it, but it seems non-trivial to implement well (given that there could be multiple levels of fragmentation or that the data could already be filtered) and I don't see an urgent need for it. I'd just leave it out for now.

**Comment by timosachsenberg** [comment]
+1 I like the name splitter better so I think it is fine

**Pull request merged**